File: fibonacci/fibonacci.go
// fib returns a function that returns
// successive Fibonacci numbers.
func generate_fibonaccis() func() uint64 {
    a, b := uint64(0), uint64(1)
	return func() uint64 {
		a, b = b, a+b
		return a
	}
}

func main() {
    fib := generate_fibonaccis()
	for j := 0; j < 10; j++ {
	    fmt.Printf("%v ", fib())
	}
}

File: fibonacci.cpp
vector<uint64_t> generate_fibonaccis(int n) {
    vector<uint64_t> fib(n);
    fib[0] = 1;
    fib[1] = 1;
    for (int i = 2; i < n; ++i) {
        fib[i] = fib[i - 2] + fib[i - 1];
    }
    return fib;
}

int main() {
    int n = 10;
    vector<uint64_t> fib = generate_fibonaccis(n);
    for (int i = 0; i < n; i++) {
        std::cout << fib[i] << " ";
    }
    return 0;
}
